### PR TITLE
Added ipynb tutorial - membrane-protein.

### LINKED ISSUE #68: Missing tutorial membrane-protein system
Add an ipynb tutorial of a membrane-protein system using the ~~[YiiP dataset](https://www.mdanalysis.org/MDAnalysisData/yiip_equilibrium.html#module-MDAnalysisData.yiip_equilibrium)~~  [NhaA dataset](https://www.mdanalysis.org/MDAnalysisData/nhaa_equilibrium.html).

To include in the tutorial:

- [x]  Introduction. 
- [x] Description of the system.
~~- [ ] Multiframe results~~.
- [x] Averaged results.
- [x] Surface, mean, Gaussian curvature plots with interpretation.
- [x] [Reference](https://www.mdanalysis.org/MDAnalysisData/yiip_equilibrium.html#fan2019).

### PR DESCRIPTION
**Pull request by ojeda-e**
This PR fixes #68 

Changes include:

- [x] Tutorial membrane-protein systems with posres.
- [x] Tutorial page updated.

### CODE REVIEW

**Comment by ojeda-e** [comment]
Here is the second tutorial ready for review @orbeckst, @lilyminium. Apologies for the delay. 

I have a question regarding the reference. I'm printing the reference with `print(NhaA.DESCR)`. I am not sure that's enough, though. Would you please suggest to me the best way to add the reference here? I considered two options.
a) Add a last section in the notebook with the reference. 
b) Add it at the beginning of the document where I give the short description of the simulation. 

Thanks!

**Comment by orbeckst** [comment]
I probably won't get to it before the end of the week. Please ping me again if you haven't heard from me by Friday. Thanks.

**Comment by lilyminium** [comment]
A nice and thorough tutorial, @ojeda-e. I'll let @orbeckst answer the question about references, since I think it's his data. I mostly found some typos but my concrete feedback on the notebook was:

* I think you can swap more-itertools for numpy
* I think the sign of the mean curvature is reversed from convention, i.e. is using the downward pointing z-axis as the normal. This isn't a documentation issue so much as a code / science one, so I raised #70. However, you are likely more familiar with membrane literature than I am -- is it the new convention in membrane MD that positive mean curvature corresponds with peaks?

**Comment by ojeda-e** [comment]
Thanks, @lilyminium for the comments. I will work on them and will replace itertools with numpy.

> I think the sign of the mean curvature is reversed from convention, i.e. is using the downward pointing z-axis as the normal. This isn't a documentation issue so much as a code / science one, so I raised Sign of mean curvature #70. However, you are likely more familiar with membrane literature than I am -- is it the new convention in membrane MD that positive mean curvature corresponds with peaks?

You're right and I appreciate you double-check this. I actually made a (terrible) mistake due to a previous mental association I had with the `bwr` colormap. This is totally my bad. The sign is the opposite, so` H>0` is valleys. The sign in the calculation of mean curvature is negative, same as in the [mean curvature function](https://github.com/MDAnalysis/membrane-curvature/blob/4bb851efdeaaeb894b7c354fd019f5f6a97079de/membrane_curvature/curvature.py#L91) in `curvature.py`.  I'm very sorry I made this mistake and I will fix it in the next push (for this tutorial). Thanks again :)

**Review by orbeckst — CHANGES_REQUESTED** [review]
Overall I like the new tutorial (and that you used NhaA ;-)). The main issue is that the protein diffuses and the tutorial does not mention or discusses this fact. I know that this remains a pain-point but there's nothing gained by hiding it. Be upfront about limitations. You don't want users thinking that the tutorial shows the membrane _around_ the protein. Instead it shows the membrane averaged over the protein diffusion (although that's probably also not well sampled). It's ok to say that this notebook demonstrates one aspect of the capabilities.

EDIT: See the nb review for specific comments.

I would be interested in seeing what it looks like when the trajectory is fitted to the protein but that can come later. For a start, just be clear about limitations.

**Comment by orbeckst** [comment]
I hope my comments on https://app.reviewnb.com/MDAnalysis/membrane-curvature/blob/issue68/docs%2Fsource%2Fpages%2FCurvature_membrane_protein_pr_all_atom.ipynb%2F/discussion actually show up...

**Comment by ojeda-e** [comment]
Thanks for the review @orbeckst. I can read your comments in the notebook although never got any notification, thanks for the link!
I made a mistake by assuming the protein was fixed. Now in retrospect, I have no clue why did I even assume so. I will work on your comments. Thanks again!

**Comment by ojeda-e** [comment]
Apologies for the delay @orbeckst.
The last push includes the suggested changes. I hope the changes are aligned with the ones you had in mind. I tried to follow all of them, please let me know if I missed anything there.

The more relevant changes are:
- Added discussion on protein diffusion (At the beginning and some other places).
- Added appendix to determine protein diffusion via MSD (Appendix 1).
- Added appendix with residue selection by leaflet. (Appendix 2).
- Updated function to generate NGL widget.

Thanks.

File: docs/source/pages/Tutorials.rst [hunk 1]
```diff
@@ -3,17 +3,39 @@
...
-   :caption: MembraneCurvature in
+   :caption: Membrane-only systems
 
    Curvature_membrane-only_systems.ipynb
-   Curvature_membrane-protein_systems_pr.ipynb
-   Curvature_membrane-protein_systems_no_pr.ipynb
 
+How to calculate curvature of a lipid bilayer of lipid composition POPC:POPE:CHOL and a
+5:4:1 ratio from a Molecular Dynamics (MD) simulation carried out using the
+Martini_ force field.
```
Comment (hmacdope) [inline docs/source/pages/Tutorials.rst:21]: Add the MARTINI citation or a link to http://cgmartini.nl  would be nice :)

File: docs/source/pages/Tutorials.rst [hunk 2]
```diff
@@ -3,17 +3,39 @@
...
+    :width: 300
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Membrane-protein systems 
+
+   Curvature_membrane_protein_pr_all_atom.ipynb
+
+MembraneCurvature to derive 2D maps of curvature induced by the sodium-proton antiporter NhaA.
+MD simulations used in this tutorial were carried out using the CHARMM36 force field.
```
Comment (hmacdope) [inline docs/source/pages/Tutorials.rst:34]: Again perhaps add a citation to the CHARMM36 FF if you mention it 👍

File: docs/source/pages/Tutorials.rst [hunk 3]
```diff
@@ -3,17 +3,39 @@
...
+    :align: center
+    :width: 300
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Membrane-protein systems 
+
+   Curvature_membrane_protein_pr_all_atom.ipynb
+
+MembraneCurvature to derive 2D maps of curvature induced by the sodium-proton antiporter NhaA.
```
Comment (hmacdope) [inline docs/source/pages/Tutorials.rst:33]:
```suggestion
Use MembraneCurvature to derive 2D maps of curvature induced by the sodium-proton antiporter NhaA.
```

**Review by hmacdope — CHANGES_REQUESTED** [review]
Initial nitpick  changes on the docs, I will review the notebook in ReviewNB.

File: docs/source/pages/Tutorials.rst (hunk 2 above)
Comment (ojeda-e) [reply docs/source/pages/Tutorials.rst:34]: Added.

**Comment by ojeda-e** [comment]
Thanks @hmacdope
I pushed the suggested changes in the `tutorials.rst`. However, I couldn't see your comments in ReviewNB. Would you please send me a link? This ReviewNB is very buggy 💀

**Comment by hmacdope** [comment]
@ojeda-e I have left comments on the notebook on the standalone review thingo at https://app.reviewnb.com/MDAnalysis/membrane-curvature/blob/issue68/docs%2Fsource%2Fpages%2FCurvature_membrane_protein_pr_all_atom.ipynb%2F/discussion

The tutorial looks and reads amazing! Mostly just little nitpicks and clarifications.

**Review by hmacdope — APPROVED** [review]
I will add @ojeda-e that I trust you to make the requisite changes so I will give my approval. You should still check with @lilyminium and @orbeckst prior to merging however.

**Pull request merged**